# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/transportbridge/MediaPipeline.cpp
# Commit: c504995adbb4
# Full Hash: c504995adbb459ffb38efbc80540b0e2dcf43ad4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Add methods to inspect buffered amount and to clear AudioPacketizer. r=padenot
#   
#   This patch also constifies members where applicable, and adds a unittest for the
#   new methods.
#   
# ==============================================================================

diff -r 605d00735b78 -r c504995adbb4 dom/media/webrtc/transportbridge/MediaPipeline.cpp
--- a/dom/media/webrtc/transportbridge/MediaPipeline.cpp	Thu Nov 05 16:42:20 2020 +0000
+++ b/dom/media/webrtc/transportbridge/MediaPipeline.cpp	Thu Nov 05 16:42:28 2020 +0000
@@ -187,8 +187,8 @@
 
     uint32_t audio_10ms = aRate / 100;
 
-    if (!mPacketizer || mPacketizer->PacketSize() != audio_10ms ||
-        mPacketizer->Channels() != aChannels) {
+    if (!mPacketizer || mPacketizer->mPacketSize != audio_10ms ||
+        mPacketizer->mChannels != aChannels) {
       // It's the right thing to drop the bit of audio still in the packetizer:
       // we don't want to send to the conduit audio that has two different
       // rates while telling it that it has a constante rate.
@@ -201,8 +201,8 @@
 
     while (mPacketizer->PacketsAvailable()) {
       mPacketizer->Output(mPacket.get());
-      mConduit->SendAudioFrame(mPacket.get(), mPacketizer->PacketSize(), aRate,
-                               mPacketizer->Channels(), 0);
+      mConduit->SendAudioFrame(mPacket.get(), mPacketizer->mPacketSize, aRate,
+                               mPacketizer->mChannels, 0);
     }
   }
 
